specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+the `spec.broker`, the Trigger can be deleted and re-created. This pattern is
+chosen to make it clear that changing the `spec.broker` is not an atomic
+operation, as it may span multiple storage systems. Changes to
+`spec.subscriber`, `spec.filter` and other fields SHOULD be permitted, as these
+could occur within a single storage system.
+
+When a Trigger becomes associated with a Broker (either due to creating the
+Trigger or the Broker), the Trigger MUST only set the `Ready` condition to
+`true` after the Broker has been configured to send all future events matching
+the `spec.filter` to the Trigger's `spec.subscriber`. The Broker MAY send some
+events to the Trigger's `spec.subscriber` prior to the Trigger's
+`Ready`condition being set to `true`. When a Trigger is deleted, the Broker MAY
+send some additional events to the Trigger's `spec.subscriber` ftor tho
+deletion.
+
+## Channel Lifecycle
+
+A Channel represents an Addressable endpoint (i.e. it has as
+`status.address.url` field) which can receive, store, and forward events to
+multiple recipients (Subscriptions). Subscriptions are associated with a Channel
+based on the `spec.channel` field on the Subscription; it is expected that the
+controller for a Channel will also control the associated Subscriptions. When
+the Channel's `Ready` condition is `true`, the Channel MUST provide a
+`status.address.url` which accepts all CloudEvents and MUST forward the received
+events to each associated Subscription whose `Ready` condition is `true`. As
+described in the [Subscription Lifecycle](#subscription-lifecycle) section, a
+Channel MAY forward events to an associated Subscription which does not
+currently have a `true` `Ready` condition, including events received by the
+Channel before the `Subscription` was created.
+
+When a Channel is created, its `spec.channelTemplate` field MUST be populated to
+indicate which of several possible Channel implementations to use. It is
+RECOMMENDED to default the `spec.channelTemplate` field on creation if it is
+unpopulated. Once created, the `spec.channelTemplate` field MUST be immutable;
+the Channel MUST be deleted and re-created to change the `spec.channelTemplate`.
+This pattern is chosen to make it clear that changing `spec.channelTemplate` is
+not an atomic operation and that any implementation would be likely to result in
+message loss during the transition.
+
+## Subscription Lifecycle
+
+The lifecycle of a Subscription is independent of that of the channel it refers
+to in its `spec.channel` field. The `spec.channel` object reference may refer to
+either an `eventing.knative.dev/v1` Channel resource, or another resource which
+meets the `spec.subscribers` and `spec.delivery` required elements in the
+Channellable duck type. If the referenced `spec.channel` does not currently
+exist or its `Ready` condition is not `true`, then the Subscription's `Ready`
+condition MUST NOT be `true`, and the reason SHOULD indicate that the
```

Comment:
MUST NOT is good, but just noting that there may be some delay if say a Channel becomes Ready->false for some reason.

Reply:
see my comment in the Broker/Trigger section on this